src/components/TransactionHistoryRows/TransactionMethod.tsx

```diff
@@ -28,6 +33,9 @@ export const TransactionMethod = ({
 }: TransactionRowProps) => {
   const translate = useTranslate()
   const txMetadata = useMemo(() => txDetails.tx.data!, [txDetails.tx.data]) // we are guaranteed to have had metadata to render this component
+  const memo = (txDetails.tx.data as { memo: string }).memo // TxMetadata typing is broken
```

Comment:
Should be happy as long as we properly discriminate cosmos `SwapMetadata` i.e:

https://github.com/shapeshift/web/blob/93b4fabdffed4ef29d9b34ff37348462d92dfca2/packages/unchained-client/src/cosmossdk/parser/types.ts#L24-L27

<img width="257" alt="image" src="https://github.com/shapeshift/web/assets/17035424/36f13329-b463-4269-bcfd-5398f8aec209">

Reply:
2309ee5ff6

Reply:
A much bigger 🧠 approach, gm ser